### PR TITLE
Add fallback mechanism when product SKU is not set

### PR DESCRIPTION
**Pull request by harrisred**
Generate custom product id when admin product id setting is set to
SKU and the product SKU field is not set.

### CODE REVIEW

File: includes/class-wc-skroutz-analytics-tracking.php [hunk 1]
```diff
@@ -139,6 +139,25 @@ private function create_action( $action, $data ) {
...
+	* product id admin setting. If the setting is SKU and the product SKU field
+	* is not set, a composite id is generated with the format wc-sa-{product id}
+	*
+	* @param WC_Product $product The purchased WC product
+	* @return string  The product id that should be reported to Analytics
+	*
+	* @since    1.0.6
+	* @access   private
+	*/
+	private function analytics_product_id( $product ) {
```
Comment (kosdiamantis) [inline includes/class-wc-skroutz-analytics-tracking.php]: [Minor] Perhaps rename it to `sa_product_id`.

**Review by kosdiamantis — APPROVED** [review]
+1 LGTM

**Comment by dkart** [comment]
+1 LGTM

File: includes/class-wc-skroutz-analytics-tracking.php [hunk 2]
```diff
@@ -139,6 +139,25 @@ private function create_action( $action, $data ) {
 	}
 
 	/**
+	* Returns the product id that should be reported to Analytics based on
+	* product id admin setting. If the setting is SKU and the product SKU field
+	* is not set, a composite id is generated with the format wc-sa-{product id}
+	*
+	* @param WC_Product $product The purchased WC product
+	* @return string  The product id that should be reported to Analytics
```
Comment (kosdiamantis) [inline includes/class-wc-skroutz-analytics-tracking.php]: `$product->id` will return an Integer 😄

**Pull request merged**